numpy/_core/src/umath/scalarmath.c.src

```diff
@@ -1262,12 +1263,9 @@ static PyObject *
              * also integers that are too large to convert to `long`), or
              * even a subclass of a NumPy scalar (currently).
              *
-             * Generally, we try dropping through to the array path here,
-             * but this can lead to infinite recursions for (c)longdouble.
+             * We drop through to the generic path here which checks for the
+             * (c)longdouble infinite recursion problem.
```

Comment:
Might make sense to just refer to this as the "infinite recursion problem", without special casing longdouble anymore. Maybe also leave a gh-XXXX reference.

Reply:
For this file only longouble is problematic, but fair, removed it and added some references.